command/agent/config/config.go

```diff
@@ -169,14 +171,253 @@ func NewConfig() *Config {
 	}
 }
 
+// Merge merges two Agent configurations.
+func (c *Config) Merge(c2 *Config) *Config {
+	if c2 == nil {
+		return c
+	}
+
+	result := NewConfig()
+
+	result.SharedConfig = c.SharedConfig
+	if c2.SharedConfig != nil {
+		result.SharedConfig = c.SharedConfig.Merge(c2.SharedConfig)
+	}
+
+	result.AutoAuth = c.AutoAuth
```

Comment:
What if e.g. some sinks are defined in one file and some in another?

Reply:
There is only one `auto_auth` block. We use the same strategy as we do for Vault itself, where we take priority to the configs we load last. We don't try and merge the contents of individual blocks.

(I know I've repeated this comment a few times, but that is hopefully easier than both of us than "see other comment")

Reply:
Huh.  That's an interesting design choice we made way back.  I agree with maintaining it for agent.